js/component.js

```diff
@@ -0,0 +1,50 @@
+/**@type {HTMLCanvasElement} */
+
+const carImg = new Image();
+carImg.src = "/images/car.png"
```

Comment:
You are saving the car image in the variable `carImg` but after when you are actually drawing the car you are using `this.image` so the car will never appear.